15314f1ddb882700dc9fabc5ca961943/update/sys_script_8bf595cc97600a10b7daf896f053affe.xml

```diff
@@ -0,0 +1,93 @@
+<?xml version="1.0" encoding="UTF-8"?><record_update sys_domain="global" table="sys_script">
+    <sys_script action="INSERT_OR_UPDATE">
+        <abort_action>false</abort_action>
+        <access>package_private</access>
+        <action_delete>false</action_delete>
+        <action_insert>true</action_insert>
+        <action_query>false</action_query>
+        <action_update>true</action_update>
+        <active>true</active>
+        <add_message>false</add_message>
+        <advanced>true</advanced>
+        <change_fields>false</change_fields>
+        <client_callable>false</client_callable>
+        <collection>x_illu2_illumio_illumio_label_dimension</collection>
+        <condition/>
+        <description/>
+        <execute_function>false</execute_function>
+        <filter_condition/>
+        <is_rest>false</is_rest>
+        <message/>
+        <name>Validate Label Dimension</name>
+        <order>100</order>
+        <priority>100</priority>
+        <rest_method/>
+        <rest_method_text/>
+        <rest_service/>
+        <rest_service_text/>
+        <rest_variables/>
+        <role_conditions/>
+        <script><![CDATA[(function executeRule(current, previous /*null when async*/ ) {
+    var valid = true;
+    var labelDimensionGr;
+    if (current.operation() == "insert") {
+        // Check if a label dimension record with the same key already exists on the table
+        labelDimensionGr = new GlideRecord("x_illu2_illumio_illumio_label_dimension");
+        labelDimensionGr.addQuery("key", current.getValue("key"));
+        labelDimensionGr.addQuery("illumio_pce_configuration", current.illumio_pce_configuration);
+        labelDimensionGr.setLimit(1);
+        labelDimensionGr.query();
+        if (labelDimensionGr.next()) {
+            valid = false;
+            gs.addErrorMessage("A label dimension with the key \"" + current.key + "\" already exists.");
+            current.setAbortAction(true);
+        }
+    }
+
+    // Check if the length of the label type initial of label dimension is not greater than 2
+    if (current.getValue("label_type_initial").length > 2) {
+        valid = false;
+        gs.addErrorMessage("The label type initial should be upto 2 characters only.");
+        current.setAbortAction(true);
+    }
+
+    // Check if a label dimension with the same label type initial already exists on the table
+    labelDimensionGr = new GlideRecord("x_illu2_illumio_illumio_label_dimension");
```

Comment:
is creating a new object needed here when you have created one in line 35?
or how about create an object and use the same across different operations?

Reply:
We’d have to create a new record as here we’re trying to clear out the previous GlideRecord session and start a new once. Hence, instead of defining a new variable we’ve reused a variable and reinitialized it.